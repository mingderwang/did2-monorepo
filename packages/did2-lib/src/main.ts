export function greet(name: string): string {
  return `Hello, ${name}! Welcome to my micromodule!`;
}

export function add(a: number, b: number): number {
  return a + b;
}

export const action2 = (name: string, times: number) => {
    let greeting = ''
    for (let i = 0; i < times; i++) {
       greeting = greeting + `Hi ${name}\n` 
    }
    return greeting
  }

console.log(action2('Alice', {times: 2}))
