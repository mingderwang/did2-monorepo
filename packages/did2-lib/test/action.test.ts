// @ts-ignore
import { test, expect } from 'bun:test';
import { action2 } from '../src/main'

test('greet function should return the correct greeting', () => {
    const greeting = action2('Alice', 2);
  expect(greeting).toMatchSnapshot();
});

