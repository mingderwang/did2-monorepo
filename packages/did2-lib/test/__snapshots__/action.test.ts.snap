// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`greet function should return the correct greeting 1`] = `
"Hi Alice
Hi Alice
"
`;
